clients/unshaded/src/main/java/tachyon/client/block/BlockStoreContext.java

```diff
@@ -55,17 +62,17 @@
    * Initializes {@link #mLocalBlockWorkerClientPool}. This method is supposed be called in a lazy
    * manner.
    */
-  private synchronized void initializeLocalBlockWorkerClientPool() {
-    NetAddress localWorkerAddress =
-        getWorkerAddress(NetworkAddressUtils.getLocalHostName(ClientContext.getConf()));
-
-    // If the local worker is not available, do not initialize the local worker client pool.
-    if (localWorkerAddress == null) {
-      mLocalBlockWorkerClientPool = null;
-    } else {
-      mLocalBlockWorkerClientPool = new BlockWorkerClientPool(localWorkerAddress);
+  private void initializeLocalBlockWorkerClientPool() {
+    if (mLocalBlockWorkerClientPoolInitialized.compareAndSet(false, true)) {
```

Comment:
consider checking `!mLocalBlockWorkerClientPoolInitialized.getAndSet(true)` so that noobs like me don't need to look up which parameter is which ;-)